azure/src/main/java/org/apache/iceberg/azure/adlsv2/ADLSLocation.java

```diff
@@ -82,4 +85,9 @@ public Optional<String> container() {
   public String path() {
     return path;
   }
+
+  /** Returns the ADLS location without any path. */
+  public String root() {
```

Comment:
Technically speaking, the trailing slash _is_ a path:

```
URI.create("abfs://container@account.dfs.core.windows.net/").getPath() = "/"
URI.create("abfs://container@account.dfs.core.windows.net").getPath() = ""
```

I wonder if it would be more correct to return the root _without_ the slash at the end.